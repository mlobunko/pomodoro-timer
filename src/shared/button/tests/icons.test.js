import React from "react";
import { shallow } from "enzyme";

import {
  faCog,
  faMinus,
  faPause,
  faPlay,
  faPlus,
  faSync,
  faVolumeOff,
  faVolumeUp
} from "@fortawesome/free-solid-svg-icons";

import {
  IconCog,
  IconMinus,
  IconPause,
  IconPlay,
  IconPlus,
  IconSync,
  IconVolumeOff,
  IconVolumeUp
} from "../icons";

describe("IconCog", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconCog />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faCog'", () => {
    const wrapper = shallow(<IconCog />);
    expect(wrapper.prop("icon")).toEqual(faCog);
  });
});

describe("IconMinus", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconMinus />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faMinus'", () => {
    const wrapper = shallow(<IconMinus />);
    expect(wrapper.prop("icon")).toEqual(faMinus);
  });
});

describe("IconPause", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconPause />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faPause'", () => {
    const wrapper = shallow(<IconPause />);
    expect(wrapper.prop("icon")).toEqual(faPause);
  });
});

describe("IconPlay", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconPlay />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faPlay'", () => {
    const wrapper = shallow(<IconPlay />);
    expect(wrapper.prop("icon")).toEqual(faPlay);
  });
});

describe("IconPlus", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconPlus />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faPlus'", () => {
    const wrapper = shallow(<IconPlus />);
    expect(wrapper.prop("icon")).toEqual(faPlus);
  });
});

describe("IconSync", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconSync />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faSync'", () => {
    const wrapper = shallow(<IconSync />);
    expect(wrapper.prop("icon")).toEqual(faSync);
  });
});

describe("IconVolumeOff", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconVolumeOff />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faVolumeOff'", () => {
    const wrapper = shallow(<IconVolumeOff />);
    expect(wrapper.prop("icon")).toEqual(faVolumeOff);
  });
});

describe("IconVolumeUp", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<IconVolumeUp />);
    expect(wrapper).toMatchSnapshot();
  });

  it("icon is 'faVolumeUp'", () => {
    const wrapper = shallow(<IconVolumeUp />);
    expect(wrapper.prop("icon")).toEqual(faVolumeUp);
  });
});
