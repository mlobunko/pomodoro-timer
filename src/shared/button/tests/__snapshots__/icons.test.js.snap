// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCog renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        512,
        512,
        Array [],
        "f013",
        "M444.788 291.1l42.616 24.599c4.867 2.809 7.126 8.618 5.459 13.985-11.07 35.642-29.97 67.842-54.689 94.586a12.016 12.016 0 0 1-14.832 2.254l-42.584-24.595a191.577 191.577 0 0 1-60.759 35.13v49.182a12.01 12.01 0 0 1-9.377 11.718c-34.956 7.85-72.499 8.256-109.219.007-5.49-1.233-9.403-6.096-9.403-11.723v-49.184a191.555 191.555 0 0 1-60.759-35.13l-42.584 24.595a12.016 12.016 0 0 1-14.832-2.254c-24.718-26.744-43.619-58.944-54.689-94.586-1.667-5.366.592-11.175 5.459-13.985L67.212 291.1a193.48 193.48 0 0 1 0-70.199l-42.616-24.599c-4.867-2.809-7.126-8.618-5.459-13.985 11.07-35.642 29.97-67.842 54.689-94.586a12.016 12.016 0 0 1 14.832-2.254l42.584 24.595a191.577 191.577 0 0 1 60.759-35.13V25.759a12.01 12.01 0 0 1 9.377-11.718c34.956-7.85 72.499-8.256 109.219-.007 5.49 1.233 9.403 6.096 9.403 11.723v49.184a191.555 191.555 0 0 1 60.759 35.13l42.584-24.595a12.016 12.016 0 0 1 14.832 2.254c24.718 26.744 43.619 58.944 54.689 94.586 1.667 5.366-.592 11.175-5.459 13.985L444.788 220.9a193.485 193.485 0 0 1 0 70.2zM336 256c0-44.112-35.888-80-80-80s-80 35.888-80 80 35.888 80 80 80 80-35.888 80-80z",
      ],
      "iconName": "cog",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconMinus renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        448,
        512,
        Array [],
        "f068",
        "M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
      ],
      "iconName": "minus",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconPause renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        448,
        512,
        Array [],
        "f04c",
        "M144 479H48c-26.5 0-48-21.5-48-48V79c0-26.5 21.5-48 48-48h96c26.5 0 48 21.5 48 48v352c0 26.5-21.5 48-48 48zm304-48V79c0-26.5-21.5-48-48-48h-96c-26.5 0-48 21.5-48 48v352c0 26.5 21.5 48 48 48h96c26.5 0 48-21.5 48-48z",
      ],
      "iconName": "pause",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconPlay renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        448,
        512,
        Array [],
        "f04b",
        "M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z",
      ],
      "iconName": "play",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconPlus renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        448,
        512,
        Array [],
        "f067",
        "M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z",
      ],
      "iconName": "plus",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconSync renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        512,
        512,
        Array [],
        "f021",
        "M440.935 12.574l3.966 82.766C399.416 41.904 331.674 8 256 8 134.813 8 33.933 94.924 12.296 209.824 10.908 217.193 16.604 224 24.103 224h49.084c5.57 0 10.377-3.842 11.676-9.259C103.407 137.408 172.931 80 256 80c60.893 0 114.512 30.856 146.104 77.801l-101.53-4.865c-6.845-.328-12.574 5.133-12.574 11.986v47.411c0 6.627 5.373 12 12 12h200.333c6.627 0 12-5.373 12-12V12c0-6.627-5.373-12-12-12h-47.411c-6.853 0-12.315 5.729-11.987 12.574zM256 432c-60.895 0-114.517-30.858-146.109-77.805l101.868 4.871c6.845.327 12.573-5.134 12.573-11.986v-47.412c0-6.627-5.373-12-12-12H12c-6.627 0-12 5.373-12 12V500c0 6.627 5.373 12 12 12h47.385c6.863 0 12.328-5.745 11.985-12.599l-4.129-82.575C112.725 470.166 180.405 504 256 504c121.187 0 222.067-86.924 243.704-201.824 1.388-7.369-4.308-14.176-11.807-14.176h-49.084c-5.57 0-10.377 3.842-11.676 9.259C408.593 374.592 339.069 432 256 432z",
      ],
      "iconName": "sync",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconVolumeOff renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        257,
        512,
        Array [],
        "f026",
        "M216.36 71.04L127.39 160H25.33c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V88.02c0-21.47-25.96-31.98-40.97-16.98z",
      ],
      "iconName": "volume-off",
      "prefix": "fas",
    }
  }
/>
`;

exports[`IconVolumeUp renders correctly 1`] = `
<StyledFontAwesomeIcon
  icon={
    Object {
      "icon": Array [
        576,
        512,
        Array [],
        "f028",
        "M215.03 71.05L126.06 160H24c-13.26 0-24 10.74-24 24v144c0 13.25 10.74 24 24 24h102.06l88.97 88.95c15.03 15.03 40.97 4.47 40.97-16.97V88.02c0-21.46-25.96-31.98-40.97-16.97zm233.32-51.08c-11.17-7.33-26.18-4.24-33.51 6.95-7.34 11.17-4.22 26.18 6.95 33.51 66.27 43.49 105.82 116.6 105.82 195.58 0 78.98-39.55 152.09-105.82 195.58-11.17 7.32-14.29 22.34-6.95 33.5 7.04 10.71 21.93 14.56 33.51 6.95C528.27 439.58 576 351.33 576 256S528.27 72.43 448.35 19.97zM480 256c0-63.53-32.06-121.94-85.77-156.24-11.19-7.14-26.03-3.82-33.12 7.46s-3.78 26.21 7.41 33.36C408.27 165.97 432 209.11 432 256s-23.73 90.03-63.48 115.42c-11.19 7.14-14.5 22.07-7.41 33.36 6.51 10.36 21.12 15.14 33.12 7.46C447.94 377.94 480 319.54 480 256zm-141.77-76.87c-11.58-6.33-26.19-2.16-32.61 9.45-6.39 11.61-2.16 26.2 9.45 32.61C327.98 228.28 336 241.63 336 256c0 14.38-8.02 27.72-20.92 34.81-11.61 6.41-15.84 21-9.45 32.61 6.43 11.66 21.05 15.8 32.61 9.45 28.23-15.55 45.77-45 45.77-76.88s-17.54-61.32-45.78-76.86z",
      ],
      "iconName": "volume-up",
      "prefix": "fas",
    }
  }
/>
`;
